### DISCUSSION TITLE
What's the per task memory usage in latest rustc and tokio?

### DISCUSSION THREAD

**Discussion by audacioustux** [tokio-rs/tokio#4678] [General]
I've ran the code mentioned in (https://github.com/tokio-rs/tokio/discussions/2650#discussion-9147) discussion in latest rust(rustc 1.59.0) and tokio(1.18.2, "full"), and it takes 1475MB for 1M tasks!!... what exactly changed in tokio or rust that made this huge difference from 342M to 1.44G?
<img width="1065" alt="Screenshot 2022-05-11 at 4 15 48 PM" src="https://user-images.githubusercontent.com/7238675/167826998-03831975-9ff5-4402-9e93-c68e8709f6d7.png">

**Comment by Darksonn** [comment]
Does downgrading Tokio change it for you? Does downgrading rustc change it for you?

**Reply by audacioustux** [reply]
<img width="1244" alt="Screenshot 2022-05-11 at 5 16 25 PM" src="https://user-images.githubusercontent.com/7238675/167837097-31e77d7d-c820-421b-96d9-3184f2783e6a.png">

yape! kinda looks like so... only downgraded tokio to 0.2.22

**Reply by Darksonn** [reply]
Would you be able to determine which version exactly the change happened in?

**Reply by audacioustux** [reply]
v1.0.0...
0.3.4
![out0 3 4](https://user-images.githubusercontent.com/7238675/167839525-9053398e-89ff-447b-ac6c-7910ad047836.png)
0.3.2
![out0 3 2](https://user-images.githubusercontent.com/7238675/167839531-49572545-b6f2-4a52-8012-75813835c808.png)
0.3.6
![out0 3 6](https://user-images.githubusercontent.com/7238675/167839507-2646e894-f4a0-40eb-9959-6ed00c4c0939.png)
1.0.0
![out1 0 0](https://user-images.githubusercontent.com/7238675/167839540-12103484-356c-4bac-aa9b-93f2b0e98620.png)
1.8.1
![out1 8 1](https://user-images.githubusercontent.com/7238675/167839548-b8b0f87a-8c5e-4002-ba07-d5a304e21e1b.png)

**Comment by shafqatevo** [comment]
I am curious - any clue or update @Darksonn?

**Reply by Darksonn** [reply]
Well, one possibility is that the size of the `Sleep` future has increased with that release.

**Comment by alexandermake** [comment]
Any progress on issue, I have the same problem

**Reply by Darksonn** [reply]
If there's still a bug we need to look into, then please open an issue instead.

**Reply by vasilakisfil** [reply]
I run the same code snippet and i get 412mb of memory usage. Not sure if it's a lot or not, but it's definitely lower than what @audacioustux reports. Running Rust 1.75 with tokio 1.38.

**Reply by wathenjiang** [reply]
Not sure. But the memory size of `Sleep` does decrease in tokio 1.38. FYI.
